1.21/src/main/java/dev/u9g/minecraftdatagenerator/generators/SoundsDataGenerator.java

```diff
@@ -0,0 +1,29 @@
+package dev.u9g.minecraftdatagenerator.generators;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import net.minecraft.registry.Registries;
+import net.minecraft.sound.SoundEvent;
+
+public class SoundsDataGenerator implements IDataGenerator {
+    public static JsonObject generateSound(SoundEvent soundEvent) {
+        JsonObject soundDesc = new JsonObject();
+
+        soundDesc.addProperty("id", Registries.SOUND_EVENT.getRawId(soundEvent) + 1); // the plus 1 is required for 1.19.2+ due to Mojang using 0 in the packet to say that you should read a string id instead.
```

Comment:
I feel like this shouldn't be on the generators end, but on the end of mineflayer.
Something like
```js
if (packet.soundId = 0) {
  readString()
} else {
  registry[packet.soundId - 1].doSomething()
}
```

At least that's how Mojang does it.